types/jasmine/index.d.ts

```diff
@@ -481,6 +497,44 @@ declare namespace jasmine {
         nothing(): void;
     }
 
+    interface AsyncMatchers<T, U> {
```

Comment:
* `T` is for what the Promise should resolve to.
* `U` is for what the Promise should reject to. This is only used in `toBeRejectedWith`.

Reply:
Users will have to specify `U` manually if they want to use it unfortunately... Inference will always result in `any` given the signature of `expectAsync`.